inspirehep/modules/migrator/tasks/records.py

```diff
@@ -315,6 +320,8 @@ def record_upsert(json):
             # Create persistent identifier.
             inspire_recid_minter(str(record.id), json)
 
+        soft_delete_record()
+
```

Comment:
Somehow this reads strange: i.e. that you would always call `soft_delete_record()` and inside it decides whether to delete it or not.

I'd suggest that you move the condition out of `soft_delete_record()` so that it's clear what will happen.